src/Tasks/FileIO/WriteLinesToFile.cs

```diff
@@ -131,36 +133,17 @@ public override bool Execute()
                             MSBuildEventSource.Log.WriteLinesToFileUpToDateStop(File.ItemSpec, false);
                         }
 
-                        if (FailIfNotIncremental)
-                        {
-                            if (Lines?.Length > 0)
-                            {
-                                Log.LogErrorWithCodeFromResources("WriteLinesToFile.ErrorReadingFile", File.ItemSpec);
-                                return false;
-                            }
-                        }
-                        else
-                        {
-                            System.IO.File.WriteAllText(File.ItemSpec, contentsAsString, encoding);
-                        }
+                        System.IO.File.WriteAllText(File.ItemSpec, contentsAsString, encoding);
                     }
                     else
                     {
-                        if (FailIfNotIncremental && Lines?.Length > 0)
+                        if (WriteOnlyWhenDifferent)
```

Comment:
I don't think I understand most of the changes in this file. Doesn't it basically mean WriteOnlyWhenDifferent no longer has any effect if Overwrite is false?

Reply:
tl;dr;: It's a desirable change.


This method is a bit of if-s madness - hard to orient self....
That's maybe the reason why it had some superfluous cases that weren't event meant to exist:

https://github.com/dotnet/msbuild/blob/9a26a2f1c569ee2f8004a356b1a138dfd93125d7/src/Tasks/Resources/Strings.resx#L2201-L2202

So this change here removes something that shouldn't event exist. So a cleanup - thumbs up for that @yuehuang010

Reply:
@Forgind, Yes, `Overwrite=false` means to append to the file.  Thus, `WriteOnlyWhenDifferent` should has no effect.  This change `FailIfNotIncremental` to only check when WriteOnlyWhenDifferent is in effect, and thus also `Overwrite=true`.

There is warning to let the user know `WriteOnlyWhenDifferent=true` is ignored.